### PR TITLE
Add the property dayComponentHeight to handle day height

### PR DESCRIPTION
**Pull request by pedrobslisboa**
### What it does:

- Add the property dayComponentHeight to handle day height, instead of having it always squared

### Before

- It was always using the width to handle height (Always squared)

### After

- It now accepts dayComponentHeight to handle it

WhatsApp Image 2021-02-10 at 14 42 29

#### This branch/PR was made based on this discussion #222 and this PR #223 

@peacechen Any problem/issue, please let me know :)

### CODE REVIEW

**Comment by peacechen** [comment]
Thanks @pedrobslisboa for updating this PR.  Please see the questions above about the size.height prop change check.

**Comment by pedrobslisboa** [comment]
> Thanks @pedrobslisboa for updating this PR. Please see the questions above about the size.height prop change check.

@peacechen I'm not seeing any question about it here. Can you elaborate?

**Comment by peacechen** [comment]
It's in the review above ☝️

**Comment by pedrobslisboa** [comment]
> It's in the review above ☝️

Screenshot 2021-02-10 at 13 11 45

Maybe your review still in progress

**Comment by peacechen** [comment]
Yes you're right, I didn't submit the review 🤣

**Comment by pedrobslisboa** [comment]
@peacechen nice questions
 For instance, we need to comprehend how it can affect this part of the code
 
 https://github.com/BugiDev/react-native-calendar-strip/blob/70e60fb7a53fc25784f666a2ff41b16febdeb3a2/src/CalendarDay.js#L151-L158
 https://github.com/BugiDev/react-native-calendar-strip/blob/70e60fb7a53fc25784f666a2ff41b16febdeb3a2/src/CalendarDay.js#L438-L442
 
I guess we should handle it either with height and width instead of generic 'size'

So we can have: 
```
containerSize: {
  width: Math.round(props.size.width),
  height: Math.round(props.size.height),
} 
```

Changing, of course, every part which deals with it.

I'm going to test it, but what do you think about it?

**Comment by peacechen** [comment]
Good catch. Those should also calculate width & height since it's no longer guaranteed to be square. However the borderRadius takes only a single value, so it will need to pick one of the dimensions.

Instead of a `size` prop in CalendarDay.js, let's pass separate width and height props. Height defaults to width, and most calculations use width by default.

File: src/CalendarDay.js [hunk 1]
```diff
@@ -90,8 +101,11 @@ class CalendarDay extends Component {
     let doStateUpdate = false;
     let hasDateChanged = prevProps.date !== this.props.date;
 
-    if ((this.props.selectedDate !== prevProps.selectedDate) || hasDateChanged) {
-      if (this.props.daySelectionAnimation.type !== "" && !this.props.scrollable) {
+    if (this.props.selectedDate !== prevProps.selectedDate || hasDateChanged) {
+      if (
+        this.props.daySelectionAnimation.type !== "" &&
```
Comment (peacechen) [inline src/CalendarDay.js:106]: Are you certain this doesn't change the evaluation logic?

**Comment by peacechen** [comment]
It looks like you're using an auto-formatter which introduced a lot of non-functional changes. I'm not completely opposed to it, but I'm concerned that someone else's auto-formatter will come back and revert those non-functional formatting changes.  Can you turn that off for these files?

**Comment by pedrobslisboa** [comment]
You're right, I have a practice to always use a shortcut to format the code, my mistake.

I've opened an issue to handle every issue like this.

https://github.com/BugiDev/react-native-calendar-strip/issues/274

**Comment by pedrobslisboa** [comment]
@peacechen Merge done and format undone

**Comment by peacechen** [comment]
Not sure if you saw this comment above
> Instead of a size prop in CalendarDay.js, let's pass separate width and height props. Height defaults to width, and most calculations use width by default.

It would be clearer to explicitly pass individual height & width props instead of a size object prop.

**Comment by pedrobslisboa** [comment]
> Not sure if you saw this comment above
> 
> > Instead of a size prop in CalendarDay.js, let's pass separate width and height props. Height defaults to width, and most calculations use width by default.
> 
> It would be clearer to explicitly pass individual height & width props instead of a size object prop.

I didn't see it, sorry. It's fixed already

**Comment by peacechen** [comment]
Thanks @pedrobslisboa this looks good. How much testing has this code gotten? We need to make sure it's solid before publishing. You can use the sample project in this repo to try different props:
https://github.com/BugiDev/react-native-calendar-strip#development-with-sample-application

**Comment by pedrobslisboa** [comment]
@peacechen I've tested it both on my android and the IOS simulator. But I am going to repeat the tests tomorrow giving you some screenshots. 
The only one I've made is the one at the description.

**Comment by pedrobslisboa** [comment]
@peacechen 

Landscape without scrollable

https://user-images.githubusercontent.com/35539594/107632721-a44d3a80-6c45-11eb-8459-a1b7d2cc091a.mp4

Portrait without scrollable

https://user-images.githubusercontent.com/35539594/107632729-a57e6780-6c45-11eb-8a47-effdd77db9b1.mp4

Landscape with scrollable

https://user-images.githubusercontent.com/35539594/107632731-a616fe00-6c45-11eb-90a7-f5bd4acaab2d.mp4

**Comment by peacechen** [comment]
Thanks for thoroughly testing 👍

**Pull request merged**